lib/net/ssh/transport/session.rb

```diff
@@ -313,7 +313,7 @@ def select_host_key_verifier(verifier)
           when :always then
             Net::SSH::Verifiers::Always.new
           else
-            if verifier.respond_to?(:verify)
+            if verifier.respond_to?(:verify) and verifier.respond_to?(:verify_signature)
```

Comment:
please use `&&` also this means it's a backward incompatible change so we need to release as 6.0

Reply:
ok sounds like breaking change. Is there any specific time frame when this release will be available?

Reply:
What  about doing soemthing like this:

```ruby
class CompatibleVerifier
  def initialize(verifier)
     @verifier
  end
  def verify_signature(key: , signature; ,hash: )
     return key.ssh_do_verify(signature, hash)
  end
end
...
if verifier.respond_to?(:verify)
  if verifyer.response_to?(:verify_signature)
    verifyer  
  else
     log "Warning: verifier without verify_signature is deprecated"
     CompatibleVerifier.new(verifier)
  end
end

```

Reply:
Make sense, will update.

Reply:
I pushed the update with all the changes you asked. Please take a look.